modules/back-end/src/Application/FeatureFlags/GetStatsByVariation.cs

```diff
@@ -0,0 +1,72 @@
+using Application.Bases;
+using Domain.FeatureFlags;
+
+namespace Application.FeatureFlags;
+
+public class GetStatsByVariation : IRequest<IEnumerable<StatsByVariationVm>>
+{
+    public Guid EnvId { get; set; }
+    public StatsByVariationFilter Filter { get; set; }
+}
+
+public class GetStatsByVariationValidator : AbstractValidator<GetStatsByVariation>
+{
+    public GetStatsByVariationValidator()
+    {
+        RuleFor(x => x.Filter.FeatureFlagKey)
+            .NotEmpty().WithErrorCode(ErrorCodes.FeatureFlagKeyIsRequired);
+
+        RuleFor(x => x.Filter.IntervalType)
+            .Must(IntervalType =>
+            {
+                return IntervalTypeEnum.Month == IntervalType || IntervalTypeEnum.Week == IntervalType ||
+                       IntervalTypeEnum.Day == IntervalType || IntervalTypeEnum.Hour == IntervalType ||
+                       IntervalTypeEnum.Minute == IntervalType;
+            }).WithErrorCode(ErrorCodes.IntervalTypeIsRequired);
```

Comment:
https://docs.fluentvalidation.net/en/latest/built-in-validators.html#enum-validator
dumb code, use `IsInEnum`

Reply:
IsInEnum is only for enum parameter, I don't think it's proper for IntervalType as it's a string? @deleteLater

Reply:
ok. how about extract this code to a method in `IntervalType` called `public bool IsDefined(string intervalType)`, then call the method here.

Reply:
and i think the error code can change to `ErrorCodes.InvalidIntervalType`

Reply:
and `IntervalTypeEnum` is not exists, this code cannot compile.

Reply:
done
didn't find IntervalTypeEnum here